tests/omp/test_omp_adamsbashforth.f90

```diff
@@ -0,0 +1,168 @@
+program test_omp_adamsbashforth
+  use iso_fortran_env, only: stderr => error_unit
+  use mpi
+
+  use m_common, only: dp, globs_t, DIR_X
+  use m_allocator, only: allocator_t, field_t
+  use m_omp_backend, only: omp_backend_t, base_backend_t
+  use m_time_integrator, only: time_intg_t
+
+  implicit none
+
+  logical :: allpass = .true.
+  real(dp), allocatable, dimension(:) :: err
+  real(dp), allocatable, dimension(:) :: norm
+  class(field_t), pointer :: u, v, w
+  class(field_t), pointer :: du, dv, dw
+
+  type(globs_t) :: globs
+  class(base_backend_t), pointer :: backend
+  class(allocator_t), pointer :: allocator
+
+  type(omp_backend_t), target :: omp_backend
+  type(allocator_t), target :: omp_allocator
+  class(time_intg_t), allocatable :: time_integrator
+
+  real(dp) :: dt0 = 0.01_dp, dt, order
+  integer :: i, j, k, istartup, ierr
+  integer :: nstep0 = 64, nstep, nrun = 4, norder = 4
+
+  ! initialize MPI
+  call MPI_Init(ierr)
+
+  ! set globs parameters
+  globs%nx = 1
+  globs%ny = 1
+  globs%nz = 1
+
+  globs%nx_loc = globs%nx
+  globs%ny_loc = globs%ny
+  globs%nz_loc = globs%nz
+
+  globs%n_groups_x = globs%ny_loc*globs%nz_loc
+  globs%n_groups_y = globs%nx_loc*globs%nz_loc
+  globs%n_groups_z = globs%nx_loc*globs%ny_loc
+
+  ! allocate object
+  omp_allocator = allocator_t(globs%nx, globs%ny, globs%nz, 1)
+  allocator => omp_allocator
+  print *, 'OpenMP allocator instantiated'
+
+  omp_backend = omp_backend_t(globs, allocator)
+  backend => omp_backend
+  print *, 'OpenMP backend instantiated'
+
+  time_integrator = time_intg_t(allocator=allocator, &
+                                backend=backend, order=norder)
+  print *, 'time integrator instantiated'
+
+  ! allocate memory
+  u => allocator%get_block(DIR_X)
+  v => allocator%get_block(DIR_X)
+  w => allocator%get_block(DIR_X)
+
+  du => allocator%get_block(DIR_X)
+  dv => allocator%get_block(DIR_X)
+  dw => allocator%get_block(DIR_X)
+
+  allocate (norm(nrun))
+
+  ! compute l2 norm for various step sizes
+  do k = 1, norder
+    time_integrator%order = k
+    dt = dt0
+    nstep = nstep0
+    do j = 1, nrun
+      ! initial condition
+      time_integrator%istep = 1
+
+      ! compute l2 norm for a given step size
+      allocate (err(nstep))
+      u%data(1, 1, 1) = 1.0_dp
+
+      ! startup
+      istartup = k - 1
+      do i = 1, istartup
+        du%data(1, 1, 1) = dahlquist_rhs(u%data(1, 1, 1))
+        call time_integrator%step(u, v, w, du, dv, dw, dt)
```

Comment:
Currently, `v`, `w`, `dv`, and `dw` are placeholder variables, only `u` and `du` are actually used for the test. I think that the `step` API should be modified to accept a flist insead, so that it can be used for an arbitrary number of variables.

Reply:
Same, maybe create an issue about it. It will be lost if it stays in the PR only